sockeye/inference.py

```diff
@@ -55,34 +55,23 @@ class InferenceModel(model.SockeyeModel):
     """
 
     def __init__(self,
-                 model_folder: str,
+                 config: model.ModelConfig,
+                 params_fname: str,
```

Comment:
what was wrong with optionally passing in the checkpoint? Conceptually it makes sense to me that the structure of the model folder is not exposed outside the InferenceModel.

Reply:
There is a couple of reasons we have the `load_models()` function by now which is essentially a factory. We need it anyway so I wanted to remove as much logic from the model constructors. But if you like the checkpoint logic in there, I can move it back in.

Reply:
Actually, I would like to leave as is since output folder is really just needed for this logic and nowhere else in `InferenceModel`. If we pull out the param_fname logic, we need only one constructor argument (fname) instead of two (checkpoint & model_folder).

Reply:
sure, it's not a big deal and I don't feel strongly about this.